app/models/link_set.rb

```diff
@@ -1,42 +1,26 @@
 class LinkSet < ActiveRecord::Base
-  include Replaceable
-  include DefaultAttributes
-  include SymbolizeJSON
-
-  validate :links_are_valid
+  has_many :links
 
-  def links=(links_hash)
-    super(links_hash || {})
-  end
+  include Replaceable
 
-private
-  def self.query_keys
-    [:content_id]
-  end
+  def hashed_links
+    links = Link.where(link_set: self)
 
-  def links_are_valid
-    # Test that the `links` attribute, if set, is a hash from strings to lists
-    # of UUIDs
-    return if links.empty?
+    links_hash = Hash.new
 
-    bad_keys = links.keys.reject { |key| link_key_is_valid?(key) }
-    if bad_keys.any?
-      errors[:links] = "Invalid link types: #{bad_keys.to_sentence}"
+    links.each do |l|
+      if links_hash.has_key?(l.link_type)
+        links_hash[l.link_type] << l.target_content_id
+      else
+        links_hash[l.link_type] = [l.target_content_id]
+      end
     end
 
-    bad_values = links.values.reject { |value|
-      value.is_a?(Array) && value.all? { |content_id|
-        UuidValidator.valid?(content_id)
-      }
-    }
-    if bad_values.any?
-      errors[:links] = "must map to lists of UUIDs"
-    end
+    links_hash.symbolize_keys
```

Comment:
each.with_object is a way of enumerating through a collection with a privately scoped object that you can add values to. In this case, you're creating a hash before the each, adding some values to it, then returning the hash. This is what each.with_object is intended to be used for.